tests/test_base.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+__author__ = "Mathematical Michael"
+__copyright__ = "Mathematical Michael"
+__license__ = "mit"
+
+
+def test_identity_mud_problem_1D(identity_problem_mud_1D):
+    # Arrange
+    D = identity_problem_mud_1D
+
+    # Act
+    mud_point = D.estimate()
+    updated_density = D._up
+    ratio = D._r
+
+    # Assert
+    assert np.round(mud_point, 1) == 0.5
+    assert np.sum(updated_density) > 0
+    assert np.mean(ratio) > 0
+
+
+def test_identity_mud_1D_with_equal_weights(identity_problem_mud_1D_equal_weights):
+    # Arrange
+    D = identity_problem_mud_1D_equal_weights
+
+    # Act
+    mud_point = D.estimate()
+    updated_density = D._up
+    ratio = D._r
+
+    # Assert
+    assert np.round(mud_point, 1) == 0.5
+    assert np.sum(updated_density) > 0
+    assert np.mean(ratio) > 0
+
+
+def test_identity_mud_1D_with_biased_weights(identity_problem_mud_1D_bias_weights):
+    # Arrange
+    D = identity_problem_mud_1D_bias_weights
```

Comment:
this just tests that we can pass non-uniform weights and still solve the mud problem, but we need another test that solves a second inverse problem using the ratio as weights from the previous.

another test to add is to mimick the usage in mud-examples that was broken and ensure that syntax will work.

Reply:
mimicking mud-example usage (setting weights when calling `set_predicted`) is added. I verified that if I reverted the change in this PR, I recreate the exact error that your previous PR caused in the other repo.